sru/release-27.14/test-migrate-user-config.sh

```diff
@@ -0,0 +1,206 @@
+    echo -e "\n\n###########################################"
+    echo "## $series: $old_version -> 27.13.4: preserve uaclient.conf changes"
+    echo "###########################################"
+    name=$(echo $series-$old_version | tr .~ -)
+
+    echo -e "\n-------------------------------------------"
+    echo "** launching container"
+    echo "-------------------------------------------"
+    lxc launch -q ubuntu-daily:$series $name
+    sleep 5
+    lxc exec $name -- apt-get update > /dev/null
+    lxc exec $name -- apt-get install debsums -y > /dev/null
+    echo "-------------------------------------------"
+
+    install_old_version $name $series $old_version
+
+    echo -e "\n-------------------------------------------"
+    echo "** make changes to uaclient.conf root"
+    echo "-------------------------------------------"
+    lxc exec $name -- sed -i "s/debug/warning/" /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show user config"
+    echo "-------------------------------------------"
+    lxc exec $name -- pro config show
+    echo "-------------------------------------------"
+
+    upgrade_to_proposed $name "NO CONFFILE PROMPT"
+
+    echo -e "\n-------------------------------------------"
+    echo "** Show uaclient.conf"
+    echo "-------------------------------------------"
+    lxc exec $name -- cat /etc/ubuntu-advantage/uaclient.conf
+    echo "-------------------------------------------"
+    echo -e "\n-------------------------------------------"
+    echo "** Show user config"
+    echo "-------------------------------------------"
+    lxc exec $name -- pro config show
+    echo "-------------------------------------------"
+
+    lxc delete --force $name
+    echo "###########################################"
+}
+
+
+# xenial
+test_normal_upgrade         xenial 27.11.3~16.04.1
+test_normal_upgrade         xenial 27.12~16.04.1
+test_normal_upgrade         xenial 27.13.1~16.04.1
+test_normal_upgrade         xenial 27.13.2~16.04.1
+test_normal_upgrade         xenial 27.13.3~16.04.1
+test_apt_news_false_upgrade  xenial 27.11.3~16.04.1
+test_apt_news_false_upgrade  xenial 27.12~16.04.1
+test_apt_news_false_upgrade  xenial 27.13.1~16.04.1
+test_apt_news_false_upgrade  xenial 27.13.2~16.04.1
+test_apt_news_false_upgrade  xenial 27.13.3~16.04.1
+test_uaclient_conf_changes_upgrade  xenial 27.11.3~16.04.1
+test_uaclient_conf_changes_upgrade  xenial 27.12~16.04.1
+test_uaclient_conf_changes_upgrade  xenial 27.13.1~16.04.1
+test_uaclient_conf_changes_upgrade  xenial 27.13.2~16.04.1
+test_uaclient_conf_changes_upgrade  xenial 27.13.3~16.04.1
```

Comment:
Can we add a test where we mess up the file permissions of `uaclient.conf` just to see the error message that will be printed during postinst ?

Reply:
Since the migration is run as root, even 000 permissions won't cause the error.

I was able to write a test that causes the error though: by making the uaclient.conf file invalid yaml.